### ISSUE TITLE
How to get phoneme's time information?

### ISSUE THREAD

**Issue by EddieWang0317** [espnet/espnet#1172]
Hello
I am working on a project trying to get the phone alignment information after CTC decoding. I have no idea of how to start. 
Can someone give me a hint about it?
THX

**Comment by sw005320** [comment]
Most of our recipes (except for TIMIT) use graphemes and you cannot obtain the phone alignment information unless you use phone transcriptions instead of grapheme transcriptions. Once you successfully make a phone system, you could obtain it from either of

1. CTC softmax output
2. attention weight

**Comment by EddieWang0317** [comment]
Hello
I try to take the CTC softmax matrix from:   
https://github.com/espnet/espnet/blob/master/espnet/nets/pytorch_backend/e2e_asr.py#L424
and use Viterbi algorithm to decoding it. But my result is quite weird.
It just like it keep output 0 and some valid symbol at the end.
e.g. ```[0,0,0,0,0,0,0,0,133,22]``` or like 
![圖片](https://user-images.githubusercontent.com/7881444/65227066-9223a700-dafa-11e9-87cb-3d3d2b0500aa.png)

The output we except is like ```[123, 123, 123, 123, 0, 66, 66, 66, 0, 72, 72]```
by the vector, we can conclude the (123) is 4 frames long. Or my understanding is wrong.

And also another question:
We found that CTC output may not identical to the final output(consider Attention and LM),
so i am wondering that can any solution to deal with this question.

**Comment by sw005320** [comment]
CTC's `0` means the blank symbol in the original paper, which can be a silence symbol or the repetition of the previous token. We cannot easily distinguish either of them.
